src/base/tristatebool.h

```diff
@@ -36,15 +36,26 @@ class TriStateBool
     static const TriStateBool False;
     static const TriStateBool True;
 
-    TriStateBool() = default;
-    TriStateBool(const TriStateBool &other) = default;
+    constexpr TriStateBool() = default;
+    constexpr TriStateBool(const TriStateBool &other) = default;
+    explicit TriStateBool(int value);  // add constexpr when using C++14
```

Comment:
IIRC, we can make it constexpr if we change its definition:
```c++
constexpr TriStateBool::TriStateBool(int value)
    : m_value {(value < 0 ? -1 : (value > 0 ? 1 : 0))}
{
}
```

Reply:
:+1:  changed to qBound

Reply:
Will it allow to make `TriStateBool &operator=(const TriStateBool &other) = default;` constexpr too?

Reply:
gcc spits error:

```
In file included from ../../src/app/cmdoptions.h:41,
                 from ../../src/app/cmdoptions.cpp:33:
../../src/base/tristatebool.h:53:29: error: defaulted declaration ‘constexpr TriStateBool& TriStateBool::operator=(const TriStateBool&) const’ does not match the expected signature
     constexpr TriStateBool &operator=(const TriStateBool &other) = default;  // add constexpr when using C++14
                             ^~~~~~~~
../../src/base/tristatebool.h:53:29: note: expected signature: ‘TriStateBool& TriStateBool::operator=(const TriStateBool&)’
../../src/base/tristatebool.h:53:29: error: explicitly defaulted function ‘constexpr TriStateBool& TriStateBool::operator=(const TriStateBool&) const’ cannot be declared as ‘constexpr’ because the implicit declaration is not ‘constexpr’:
make[1]: *** [Makefile:2743: cmdoptions.o] Error 1
```